aip-console-tools-cli/src/test/java/com/castsoftware/aip/console/tools/DeliverVersionCommandIntegrationTest.java

```diff
@@ -0,0 +1,207 @@
+package com.castsoftware.aip.console.tools;
+
+import com.castsoftware.aip.console.tools.commands.DeliverVersionCommand;
+import com.castsoftware.aip.console.tools.core.dto.jobs.JobRequestBuilder;
+import com.castsoftware.aip.console.tools.core.dto.jobs.JobState;
+import com.castsoftware.aip.console.tools.core.dto.jobs.JobStatusWithSteps;
+import com.castsoftware.aip.console.tools.core.exceptions.ApplicationServiceException;
+import com.castsoftware.aip.console.tools.core.exceptions.JobServiceException;
+import com.castsoftware.aip.console.tools.core.exceptions.PackagePathInvalidException;
+import com.castsoftware.aip.console.tools.core.exceptions.UploadException;
+import com.castsoftware.aip.console.tools.core.utils.Constants;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+import picocli.CommandLine;
+
+import java.io.File;
+import java.util.Date;
+import java.util.function.Function;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE, classes = {AipConsoleToolsCliIntegrationTest.class})
+@ActiveProfiles(TestConstants.PROFILE_INTEGRATION_TEST)
+public class DeliverVersionCommandIntegrationTest extends AipConsoleToolsCliBaseTest {
+    @Autowired
+    private DeliverVersionCommand deliverVersionCommand;
+
+    @Override
+    protected void cleanupTestCommant() {
+        // ===================================
+        //command not recreated between test.
+        //So just clear the command as if it was brand newly created
+        //The best way is to run java -jar .\aip-console-tools-cli.jar add ...
+        // So that app creates new instances of commands.
+        // Still this woks fine renewing parameters values each time.
+        // Here only String types, but each test should set velues to requested ones
+        // ===================================
+        resetSharedOptions(deliverVersionCommand.getSharedOptions());
+        deliverVersionCommand.setApplicationName(null);
+        deliverVersionCommand.setBackupName(null);
+        deliverVersionCommand.setDomainName(null);
+        deliverVersionCommand.setFilePath(null);
+        deliverVersionCommand.setNodeName(null);
+        deliverVersionCommand.setVersionName(null);
+    }
+
+    @Test
+    public void testDeliverVersionCommand_UsageHelp() {
```

Comment:
Please remove this test, same remark regarding maintainability and updating usage help text content.

Reply:
All of them will get removed